pkg/docker/docker.go

```diff
@@ -472,17 +479,40 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) (err error) {
 		}
 	}
 
-	glog.V(2).Infof("Waiting for container")
-	exitCode, err := d.client.WaitContainer(container.ID)
-	wg.Wait()
-	if err != nil {
-		return err
-	}
-	glog.V(2).Infof("Container exited")
+	if opts.TargetImage {
+		exposedPorts := make([]string, 0)
+		for port := range imageMetadata.Config.ExposedPorts {
+			exposedPorts = append(exposedPorts, port.Port())
+		}
 
-	if exitCode != 0 {
-		return errors.NewContainerError(container.Name, exitCode, "")
+		glog.V(0).Infof("\n\n\n\n\nThe image %s has been started in container %s as a result of the --run=true option.  The container's stdout/stderr will be redirected to this command's glog output to help you validate its behavior.  You can also inspect the container with docker commands if you like.  If the container is set up to stay running, you will have to Ctrl-C to exit this command, which should also stop the container %s.  This particular invocation attempts to run on the exposed ports %+v \n\n\n\n\n", image, container.ID, container.ID, exposedPorts)
```

Comment:
Hmm ... unfortunately, the NetworkSettings are coming up `nil` in my testing.  Perhaps the Container ref returned by CreateContainer is not updated when other operations like StartContainer are done later on.

I could look into exposing https://github.com/openshift/source-to-image/blob/master/Godeps/_workspace/src/github.com/fsouza/go-dockerclient/container.go#L57-L69, which if called after StartContainer, perhaps the APIContainer / APIPort at https://github.com/openshift/source-to-image/blob/master/Godeps/_workspace/src/github.com/fsouza/go-dockerclient/container.go#L32-L52 could get us what you want.